homeassistant/components/osc/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""The OSC (Open Sound Control) integration."""
+import logging
+
+from pythonosc.udp_client import SimpleUDPClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+# Define the service schema
+SERVICE_OSC_SEND_SCHEMA = vol.Schema(
+    {
+        vol.Required("entity_id"): cv.string,
```

Comment:
`cv.entity_id` instead of `cv.string` to get some automatic validation on the entity ID

Reply:
You should also look at entity selectors to limit this even further. I am assuming you only want entity's from the integration as input, a selector can limit the user to that. Example: https://github.com/home-assistant/core/blob/77b421887befe1478fe40798e301d25721cd30c4/homeassistant/components/min_max/config_flow.py#L34C1-L39